src/send/SendSelectRecipient.tsx

```diff
@@ -138,32 +140,29 @@ function SendSelectRecipient() {
     setSearchQuery(searchInput)
   }, SEARCH_THROTTLE_TIME)
 
-  const [shouldImportContacts, setShouldImportContacts] = useState(false)
+  const [showContacts, setShowContacts] = useState(false)
 
   const recentRecipients = useSelector((state) => state.send.recentRecipients)
+  const contactsCache = useSelector(phoneRecipientCacheSelector)
+  const contactRecipients = useMemo(
+    () => sortRecipients(Object.values(contactsCache)),
+    [contactsCache]
+  )
   const showGetStarted = !recentRecipients.length
 
-  const defaultCountryCode = useSelector(defaultCountryCodeSelector)
-
   const { recipientVerificationStatus, recipient, setSelectedRecipient } =
     useFetchRecipientVerificationStatus()
 
   const dispatch = useDispatch()
 
-  useAsync(async () => {
-    if (shouldImportContacts) {
-      // TODO (ACT-949): Currently if a user does not have a phone number linked, this silently fails
-      const permissionGranted = await requestContactsPermission()
-      if (permissionGranted) {
-        dispatch(importContacts())
-      }
-      setShouldImportContacts(false)
+  const onPressContacts = async () => {
+    ValoraAnalytics.track(SendEvents.send_select_recipient_contacts)
+    const permissionGranted = await requestContactsPermission()
```

Comment:
Just to check, on iOS, the behavior here is:
* If permissions have already been granted, this does nothing and we just show contacts.
* If permissions have been previously denied, this will not show a native modal, and just silently return false.
* If permissions have been neither granted nor denied previously, this will show a native modal.

Is that understanding correct? How do we differentiate between the second bullet and the case where the user rejects the native modal in the third case?

On Android, we can call `requestContactsPermission` as many times as we like, and the native modal will appear each time, right?

Reply:
yes your understanding on iOS behavior is correct. I think we have an API that returns "authorized" for 1, "declined" for 2 and "undefined" for 3.
For Android, I am still digging into this but I think if the user denies multiple times, it gets into a "never_ask_again" state, in which case `requestContactsPermission` will silently fail like the iOS denied case.